### ISSUE TITLE
Moving existing Ingress objects into a group leaves orphan AWS ALBs behind

### ISSUE THREAD

**Issue by jbg** [kubernetes-sigs/aws-load-balancer-controller#2028]
I had several Ingress objects corresponding to individual AWS ALBs.

I added a matching `alb.ingress.kubernetes.io/group.name: my-group-name` annotation to all of them, and the controller correctly provisioned a new ALB for the group.

However, the individual ALBs were not deleted and need to be manually cleaned up.

**Comment by kishorj** [comment]
@jbg, what is the controller version you are running? Do you see any errors in the controller logs?

I tried on v2.2.0, and wasn't able to reproduce the issue.

**Comment by jbg** [comment]
v2.2.0. I did it again on another two EKS clusters and got exactly the same behaviour on both.

**Comment by kishorj** [comment]
do you see any errors in the controller logs?

**Comment by M00nF1sh** [comment]
@jbg 
would you share the cluster ARN to us (with your reproduce in the cluster )? `yyyng AT amazon.com` or `joshikis AT amazon.com `

**Comment by M00nF1sh** [comment]
@jbg 
Does this issue still persist?
If so, feel free to open a new issue to us 😄

**Comment by jbg** [comment]
Yes, this still persists (we've done this type of migration multiple times in production on multiple clusters and we've always had to delete the orphaned ALBs afterwards).

Most recently, we reproduced it yesterday: We had four existing Ingress objects. We added an annotation to each of them with the same value for `alb.ingress.kubernetes.io/group.name`. The controller created the new ALB for the group and associated all the Ingress objects with it. However, the old ALBs corresponding to each Ingress are still there and must be manually deleted.

The controller doesn't log any errors, in fact it doesn't log anything related to deleting ALBs at all. Can anyone point me at the code that should handle this?
